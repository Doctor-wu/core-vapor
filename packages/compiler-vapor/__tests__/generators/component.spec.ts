import { compile } from '@vue/compiler-vapor'

describe('generate component', () => {
  test('generate single root component (without props)', () => {
    const { code } = compile(`<Comp/>`)
    expect(code).toMatchSnapshot()
  })

  test('generate single root component (with props)', () => {
    const { code } = compile(`<Comp :foo="foo"/>`)
    expect(code).toMatchSnapshot()
  })

  test('generate multi root component', () => {
    const { code } = compile(`<Comp/>123`)
    expect(code).toMatchSnapshot()
  })
})
