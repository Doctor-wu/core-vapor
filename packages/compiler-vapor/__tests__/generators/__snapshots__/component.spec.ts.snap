// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`generate component > generate multi root component 1`] = `
"import { resolveComponent as _resolveComponent, createComponent as _createComponent, template as _template } from 'vue/vapor';
const t0 = _template("123")

export function render(_ctx) {
  const n1 = t0()
  const n0 = _createComponent(_resolveComponent("Comp"))
  return [n0, n1]
}"
`;

exports[`generate component > generate single root component (with props) 1`] = `
"import { resolveComponent as _resolveComponent, createComponent as _createComponent } from 'vue/vapor';

export function render(_ctx) {
  const n0 = _createComponent(_resolveComponent("Comp"), [{
    foo: () => (foo)
  }], true)
  return n0
}"
`;

exports[`generate component > generate single root component (without props) 1`] = `
"import { resolveComponent as _resolveComponent, createComponent as _createComponent } from 'vue/vapor';

export function render(_ctx) {
  const n0 = _createComponent(_resolveComponent("Comp"), null, true)
  return n0
}"
`;
